### PR TITLE
[UNTIL-11629]Scanner list improvements

### PR DESCRIPTION
**Pull request by shekar-allam**

### CODE REVIEW

**Review by djordjeh — APPROVED** [review]

File: scan-engine/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -11,4 +11,5 @@
     <string name="loading">Loading</string>
     <string name="pairing_title">Barcode pairing</string>
     <string name="pairing_successful">Pairing successful</string>
+    <string name="pairing_instruction">Scan barcode to pair your cordless scanner to this app.</string>
```
[the remark this answers is not in the document]
Comment (djordjeh) [reply scan-engine/src/main/res/values/strings.xml:14]: we need german translation as well

File: scan-engine/src/main/java/de/tillhub/scanengine/data/ScannerEvent.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+sealed class ScannerEvent {
+    sealed class External : ScannerEvent() {
+        data object NotConnected : External()
+        data object Connected : External()
+    }
+    sealed class Camera : ScannerEvent() {
+        class InProgress(val scanKey: String?) : Camera()
+        data object Canceled : Camera()
+    }
+    class Success(
```
Comment (djordjeh) [inline scan-engine/src/main/java/de/tillhub/scanengine/data/ScannerEvent.kt:11]: We said ScanResult

**Review by djordjeh — CHANGES_REQUESTED** [review]

File: scan-engine/proguard-rules.pro [hunk 3]
```diff
@@ -27,8 +27,8 @@
 -keep class de.tillhub.scanengine.ScanEngine$Companion { *; }
 -keep class de.tillhub.scanengine.common.SingletonHolder { *; }
 
--keep class de.tillhub.scanengine.data.ScanEvent { *; }
--keep class de.tillhub.scanengine.data.ScanEvent* { *; }
+-keep class de.tillhub.scanengine.data.ScannerEvent { *; }
+-keep class de.tillhub.scanengine.data.ScannerEvent* { *; }
```
Comment (djordjeh) [inline scan-engine/proguard-rules.pro:31]: You might need more here now. Please test with Pos Go and let me know.

**Review by djordjeh — CHANGES_REQUESTED** [review]

File: scan-engine/proguard-rules.pro (hunk 3 above)
Comment (shekar-allam) [reply scan-engine/proguard-rules.pro:31]: this is enough , able to see these on posgo

**Review by djordjeh — APPROVED** [review]

**Pull request merged**